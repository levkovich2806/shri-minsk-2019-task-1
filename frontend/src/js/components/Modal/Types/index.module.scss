.types {
  display: flex;

  align-items: center;
  margin-bottom: 10px;

  &__title {
    font-size: 16px;
    line-height: 22px;
    font-weight: bold;
  }

  &__item {
    margin-left: 10px;

    input,
    label {
      cursor: pointer;
    }
  }
}
