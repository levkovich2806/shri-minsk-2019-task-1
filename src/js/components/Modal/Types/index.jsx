import React from "react";
import styles from "./index.module.scss";

function Types(props) {
  return (
    <div className={styles.types}>
      <div className={styles.types__title}>Тип: </div>
      <div className={styles.types__item}>
        <input
          onChange={props.onChange}
          type="radio"
          id="text"
          name="type"
          value="text"
          defaultChecked
        />
        <label htmlFor="text">Текст</label>
      </div>
      <div className={styles.types__item}>
        <input
          onChange={props.onChange}
          type="radio"
          id="image"
          name="type"
          value="image"
        />
        <label htmlFor="image">Изображение</label>
      </div>
      <div className={styles.types__item}>
        <input
          onChange={props.onChange}
          type="radio"
          id="list"
          name="type"
          value="list"
        />
        <label htmlFor="list">Список</label>
      </div>
    </div>
  );
}

export default Types;
