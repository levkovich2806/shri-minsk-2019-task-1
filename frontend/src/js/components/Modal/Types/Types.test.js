import React from 'react';
import { shallow, mount } from 'enzyme';

import Types from './index';

describe('Types', () => {

  it('Should match snapshot Types', () => {
    const wrapper = shallow(<Types type="image" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Types - check correct checked radio", () => {

    const type = "image";

    const wrapper = mount(<Types type={"image"} onChange={() => { }} />);

    const input = wrapper.find("input[type='radio']");
    input.forEach(item => {
      const { checked, value } = item.get(0).props;
      if (type === value) {
        expect(checked).toBe(true);
      }
    });
  });

});
