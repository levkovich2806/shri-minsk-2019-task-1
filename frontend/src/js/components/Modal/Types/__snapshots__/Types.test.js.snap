// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Types Should match snapshot Types 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Types
    type="image"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="types__title"
        >
          Тип: 
        </div>,
        <div
          className="types__item"
        >
          <input
            checked={false}
            id="text"
            name="type"
            type="radio"
            value="text"
          />
          <label
            htmlFor="text"
          >
            Текст
          </label>
        </div>,
        <div
          className="types__item"
        >
          <input
            checked={true}
            id="image"
            name="type"
            type="radio"
            value="image"
          />
          <label
            htmlFor="image"
          >
            Изображение
          </label>
        </div>,
        <div
          className="types__item"
        >
          <input
            checked={false}
            id="list"
            name="type"
            type="radio"
            value="list"
          />
          <label
            htmlFor="list"
          >
            Список
          </label>
        </div>,
      ],
      "className": "types",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Тип: ",
          "className": "types__title",
        },
        "ref": null,
        "rendered": "Тип: ",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              id="text"
              name="type"
              type="radio"
              value="text"
            />,
            <label
              htmlFor="text"
            >
              Текст
            </label>,
          ],
          "className": "types__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "text",
              "name": "type",
              "onChange": undefined,
              "type": "radio",
              "value": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Текст",
              "htmlFor": "text",
            },
            "ref": null,
            "rendered": "Текст",
            "type": "label",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={true}
              id="image"
              name="type"
              type="radio"
              value="image"
            />,
            <label
              htmlFor="image"
            >
              Изображение
            </label>,
          ],
          "className": "types__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "id": "image",
              "name": "type",
              "onChange": undefined,
              "type": "radio",
              "value": "image",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Изображение",
              "htmlFor": "image",
            },
            "ref": null,
            "rendered": "Изображение",
            "type": "label",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              id="list"
              name="type"
              type="radio"
              value="list"
            />,
            <label
              htmlFor="list"
            >
              Список
            </label>,
          ],
          "className": "types__item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "list",
              "name": "type",
              "onChange": undefined,
              "type": "radio",
              "value": "list",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Список",
              "htmlFor": "list",
            },
            "ref": null,
            "rendered": "Список",
            "type": "label",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="types__title"
          >
            Тип: 
          </div>,
          <div
            className="types__item"
          >
            <input
              checked={false}
              id="text"
              name="type"
              type="radio"
              value="text"
            />
            <label
              htmlFor="text"
            >
              Текст
            </label>
          </div>,
          <div
            className="types__item"
          >
            <input
              checked={true}
              id="image"
              name="type"
              type="radio"
              value="image"
            />
            <label
              htmlFor="image"
            >
              Изображение
            </label>
          </div>,
          <div
            className="types__item"
          >
            <input
              checked={false}
              id="list"
              name="type"
              type="radio"
              value="list"
            />
            <label
              htmlFor="list"
            >
              Список
            </label>
          </div>,
        ],
        "className": "types",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Тип: ",
            "className": "types__title",
          },
          "ref": null,
          "rendered": "Тип: ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                id="text"
                name="type"
                type="radio"
                value="text"
              />,
              <label
                htmlFor="text"
              >
                Текст
              </label>,
            ],
            "className": "types__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "text",
                "name": "type",
                "onChange": undefined,
                "type": "radio",
                "value": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Текст",
                "htmlFor": "text",
              },
              "ref": null,
              "rendered": "Текст",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={true}
                id="image"
                name="type"
                type="radio"
                value="image"
              />,
              <label
                htmlFor="image"
              >
                Изображение
              </label>,
            ],
            "className": "types__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "id": "image",
                "name": "type",
                "onChange": undefined,
                "type": "radio",
                "value": "image",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Изображение",
                "htmlFor": "image",
              },
              "ref": null,
              "rendered": "Изображение",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                id="list"
                name="type"
                type="radio"
                value="list"
              />,
              <label
                htmlFor="list"
              >
                Список
              </label>,
            ],
            "className": "types__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "list",
                "name": "type",
                "onChange": undefined,
                "type": "radio",
                "value": "list",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Список",
                "htmlFor": "list",
              },
              "ref": null,
              "rendered": "Список",
              "type": "label",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
